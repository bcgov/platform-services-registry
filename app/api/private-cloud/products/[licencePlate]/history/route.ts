import prisma from '@/core/prisma';
import { z } from 'zod';
import createApiHandler from '@/core/api-handler';
import { NotFoundResponse, OkResponse } from '@/core/responses';

const pathParamSchema = z.object({
  licencePlate: z.string(),
});

const apiHandler = createApiHandler({
  roles: ['user'],
  validations: { pathParams: pathParamSchema },
});

export const GET = apiHandler(async ({ pathParams, session }) => {
  const { licencePlate } = pathParams;

  const requests = await prisma.privateCloudRequest.findMany({
    where: {
      licencePlate,
    },
    orderBy: {
      created: 'desc',
    },
    session: session as never,
  });

  if (!requests) {
    return NotFoundResponse('No requests found for this licence plate.');
  }

  return OkResponse(requests);
});
