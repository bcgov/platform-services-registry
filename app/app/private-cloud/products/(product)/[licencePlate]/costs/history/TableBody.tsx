'use client';

import { Table } from '@mantine/core';
import { YearlyCostDataWithMonthName } from '@/helpers/product';

export interface TableProps {
  data: YearlyCostDataWithMonthName[];
  currentYear: string;
}

export default function TableBody({ data, currentYear }: TableProps) {
  const rows = data.length ? (
    data.map((resource, index) => (
      <Table.Tr key={index}>
        <Table.Td>
          {resource.month} {currentYear}
        </Table.Td>
        <Table.Td>{resource.cpuCost}</Table.Td>
        <Table.Td>{resource.storageCost}</Table.Td>
        <Table.Td>{resource.totalCost}</Table.Td>
      </Table.Tr>
    ))
  ) : (
    <Table.Tr>
      <Table.Td colSpan={5} className="italic">
        No Cost History Found
      </Table.Td>
    </Table.Tr>
  );

  return (
    <Table.ScrollContainer minWidth={800}>
      <Table verticalSpacing="sm" striped withColumnBorders withTableBorder horizontalSpacing="md">
        <Table.Thead>
          <Table.Tr>
            <Table.Th>Month</Table.Th>
            <Table.Th>CPU Usage Cost (CAD)</Table.Th>
            <Table.Th>Storage Usage Cost (CAD)</Table.Th>
            <Table.Th>Total Cost (CAD)</Table.Th>
          </Table.Tr>
        </Table.Thead>
        <Table.Tbody>{rows}</Table.Tbody>
      </Table>
    </Table.ScrollContainer>
  );
}
