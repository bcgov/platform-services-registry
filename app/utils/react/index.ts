export * from './hooks';
export * from './search-params';
