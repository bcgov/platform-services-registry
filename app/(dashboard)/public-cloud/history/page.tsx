import Table from '@/components/table/Table';
import TableBody from '@/components/table/TableBody';
import { publicCloudRequestsPaginated } from '@/queries/paginated/public-cloud';
import { publicCloudRequestDataToRow } from '@/components/table/helpers/rowMapper';
import { getServerSession } from 'next-auth/next';
import { authOptions } from '@/app/api/auth/options';
import { redirect } from 'next/navigation';
import { userInfo } from '@/queries/user';

const headers = [
  { field: 'created', headerName: 'Date' },
  { field: 'name', headerName: 'Name' },
  { field: 'type', headerName: 'Type' },
  { field: 'status', headerName: 'Status' },
  { field: 'ministry', headerName: 'Ministry' },
  { field: 'cluster', headerName: 'Provider' },
  // { field: 'projectOwner', headerName: 'Project Owner' },
  // { field: 'technicalLeads', headerName: 'Technical Leads' },
  { field: 'licencePlate', headerName: 'Licence Plate' },
];

export default async function RequestsTable({
  searchParams,
}: {
  searchParams: {
    search: string;
    page: number;
    pageSize: number;
    ministry: string;
    provider: string;
    active: boolean;
  };
}) {
  // Authenticate the user
  const session = await getServerSession(authOptions);

  if (!session) {
    redirect('/login?callbackUrl=/private-cloud/products/all');
  }

  const { search, page, pageSize, ministry, provider, active } = searchParams;
  const { userEmail, ministryRoles } = userInfo(session.user.email, session.user.roles);

  // If a page is not provided, default to 1
  const currentPage = typeof searchParams.page === 'string' ? +page : 1;
  const defaultPageSize = 10;

  // If not an admin, we need to provide the user's email to the query

  const { data, total } = await publicCloudRequestsPaginated(
    +pageSize || defaultPageSize,
    currentPage,
    search,
    ministry,
    provider,
    userEmail,
    ministryRoles,
    active,
  );

  const rows = data.map(publicCloudRequestDataToRow);

  return (
    <Table
      title="Requests for Public Cloud Landing Zones"
      description="These are the submitted requests for your products in the Public Cloud Landing Zones"
      tableBody={<TableBody headers={headers} rows={rows} />}
      total={total}
      currentPage={currentPage}
      pageSize={pageSize || defaultPageSize}
    />
  );
}
