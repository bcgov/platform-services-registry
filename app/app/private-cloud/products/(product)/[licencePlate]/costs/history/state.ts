import { proxy } from 'valtio';
import { deepClone } from 'valtio/utils';
import { YearlyCostPrivateCloudFilterBody } from '@/validation-schemas/private-cloud';

const initialValue = {
  year: new Date().getFullYear().toString(),
};

export const pageState = proxy<YearlyCostPrivateCloudFilterBody>(deepClone(initialValue));
