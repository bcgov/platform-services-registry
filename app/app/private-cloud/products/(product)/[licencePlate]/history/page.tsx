'use client';

import { useQuery } from '@tanstack/react-query';
import { useEffect, useState } from 'react';
import { z } from 'zod';
import PrivateHistoryItem from '@/components/history/PrivateHistoryItem';
import { GlobalRole } from '@/constants';
import createClientPage from '@/core/client-page';
import { getPrivateCloudProductRequests } from '@/services/backend/private-cloud/products';

const pathParamSchema = z.object({
  licencePlate: z.string(),
});

const privateCloudProductHistory = createClientPage({
  roles: [GlobalRole.User],
  validations: { pathParams: pathParamSchema },
});
export default privateCloudProductHistory(({ getPathParams, session }) => {
  const [pathParams, setPathParams] = useState<z.infer<typeof pathParamSchema>>();

  useEffect(() => {
    getPathParams().then((v) => setPathParams(v));
  }, []);

  const { licencePlate = '' } = pathParams ?? {};

  const {
    data: requests,
    isLoading: requestsLoading,
    isError: requestsIsError,
    error: requestsError,
  } = useQuery({
    queryKey: ['requests', licencePlate],
    queryFn: () => getPrivateCloudProductRequests(licencePlate),
    enabled: !!licencePlate,
  });

  if (!requests) return null;

  return (
    <>
      {requests.map((request) => (
        <PrivateHistoryItem key={request.id} {...request} />
      ))}
    </>
  );
});
