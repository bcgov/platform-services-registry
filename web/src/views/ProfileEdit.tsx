//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import { useKeycloak } from '@react-keycloak/web';
import React, { useEffect, useState } from 'react';
import { Link as RouterLink, Redirect } from 'react-router-dom';
import { Box, Flex, Text } from 'rebass';
import Icon from '../components/common/UI/Icon';
import { ShadowBox } from '../components/common/UI/ShadowContainer';
import ContactCard from '../components/profileEdit/ContactCard';
import ContactCardEdit from '../components/profileEdit/ContactCardEdit';
import ProjectCard from '../components/profileEdit/ProjectCard';
import ProjectCardEdit from '../components/profileEdit/ProjectCardEdit';
import QuotaCard from '../components/profileEdit/QuotaCard';
import QuotaCardEdit from '../components/profileEdit/QuotaCardEdit';
import { PROFILE_EDIT_VIEW_NAMES, RESPONSE_STATUS_CODE, ROUTE_PATHS } from '../constants';
import useInterval from '../hooks/useInterval';
import useRegistryApi from '../hooks/useRegistryApi';
import theme from '../theme';
import { CNQuotaOptions, Namespace, QuotaSizeSet } from '../types';
import getProfileStatus from '../utils/getProfileStatus';
import { promptErrToastWithText } from '../utils/promptToastHelper';
import { getCurrentQuotaOptions, getCurrentQuotaSize, getLicensePlate, getProfileContacts, getProfileMinistry, isProfileProvisioned } from '../utils/transformDataHelper';
const txtForQuotaEdit = `All quota increase requests require Platform Services Team's approval. Please contact the Platform Admins (@cailey.jones, @patrick.simonian or @shelly.han) in RocketChat BEFORE submitting the request to provide justification for the increased need of Platform resources (i.e. historic data showing increased CPU/RAM consumption).`;

interface IProfileEditProps {
    openBackdropCB: () => void;
    closeBackdropCB: () => void;
};

const ProfileEdit: React.FC<IProfileEditProps> = (props) => {
    const api = useRegistryApi();
    const { keycloak } = useKeycloak();

    const profileStatus = getProfileStatus();

    // @ts-ignore
    const { match: { params: { profileId, viewName } }, openBackdropCB, closeBackdropCB } = props;

    const [initialRender, setInitialRender] = useState(true);
    const [unauthorizedToAccess, setUnauthorizedToAccess] = useState(false);
    const [profileJson, setProfileJson] = useState<any>({});
    const [contactJson, setContactJson] = useState<any>({});
    const [ministry, setMinistry] = useState<any>([]);
    const [provisionedStatus, setProvisionedStatus] = useState<any>();

    const [isProvisioned, setIsProvisioned] = useState<boolean>(false);
    const [namespacesJson, setNamespacesJson] = useState<Namespace[]>([]);
    const [quotaSize, setQuotaSize] = useState<QuotaSizeSet | ''>('');
    const [licensePlate, setLicensePlate] = useState<string>('');
    const [cnQuotaOptionsJson, setCnQuotaOptionsJson] = useState<CNQuotaOptions[]>([]);
    const [quotaOptions, setQuotaOptions] = useState<QuotaSizeSet[]>([]);
    const [quotaSubmitRefresh, setQuotaSubmitRefresh] = useState<any>(0);

    const [pendingEditRequest, setPendingEditRequest] = useState(true);

    const handleQuotaSubmitRefresh = () => {
        setQuotaSubmitRefresh(quotaSubmitRefresh + 1);
    };

    useEffect(() => {
        async function wrap() {
            openBackdropCB();
            try {
                const profileDetails = await api.getProfileByProfileId(profileId);
                const ministryDetails = await api.getMinistry();
                setMinistry(ministryDetails.data);

                profileDetails.data = { ...profileDetails.data, ...getProfileMinistry(ministryDetails.data, profileDetails.data) };
                setProfileJson(profileDetails.data);

                const contactDetails = await api.getContactsByProfileId(profileId);
                contactDetails.data = { ...getProfileContacts(contactDetails.data) };
                setContactJson(contactDetails.data);

                const namespaces = await api.getNamespacesByProfileId(profileId);
                const cnQuotaOptions = await api.getCNQuotaOptionsByProfileId(profileId);


                setIsProvisioned(isProfileProvisioned(namespaces.data));
                setNamespacesJson(namespaces.data);
                setCnQuotaOptionsJson(cnQuotaOptions.data);

                const isProvisioned = isProfileProvisioned(namespaces.data);
                setProvisionedStatus(isProvisioned);

                const isPendingEditRequest = await profileStatus.getPendingEditRequest(api, profileId)
                setPendingEditRequest(isPendingEditRequest);

            } catch (err) {
                if (err.response && err.response.status && err.response.status === RESPONSE_STATUS_CODE.UNAUTHORIZED) {
                    setUnauthorizedToAccess(true);
                } else {
                    // when api returns 500 or queried profile entry does not exist
                    promptErrToastWithText('Something went wrong');
                }
            }
            setInitialRender(false);
            closeBackdropCB();
        }
        wrap();
        // eslint-disable-next-line
    }, [keycloak]);

    useEffect(() => {
        async function wrap() {
            if (namespacesJson.length === 0 || cnQuotaOptionsJson.length === 0) { return; }
            try {
                // @ts-ignore
                setLicensePlate(getLicensePlate(namespacesJson));
                // @ts-ignore
                setQuotaSize(getCurrentQuotaSize(namespacesJson));
                // @ts-ignore
                setQuotaOptions(getCurrentQuotaOptions(cnQuotaOptionsJson, getCurrentQuotaSize(namespacesJson)));
            } catch (err) {
                promptErrToastWithText(err.message);
            }
        }
        wrap();
    }, [namespacesJson, cnQuotaOptionsJson, quotaSubmitRefresh]);

    useInterval(() => {
        async function wrap() {
            const isPendingEditRequest = await profileStatus.getPendingEditRequest(api, profileId)
            setPendingEditRequest(isPendingEditRequest);
        }
        wrap();
    }, 1000 * 30);

    if (initialRender) {
        return null;
    }

    if (unauthorizedToAccess) {
        return <Redirect to={ROUTE_PATHS.NOT_FOUND} />;
    }

    if (viewName === PROFILE_EDIT_VIEW_NAMES.OVERVIEW) {
        return (
            <>
                <Box sx={{
                    display: 'grid',
                    gridGap: 4
                }}>
                    <ShadowBox p={5} style={{ position: 'relative' }}>
                        <Text as="h1">
                            {profileJson.name}
                        </Text>
                        <Box>
                            <Flex p={3} mt={4} bg={theme.colors.bcblue} style={{ position: 'relative' }}>
                                <Text as="h3" color={theme.colors.contrast} mx={2} >
                                    Project Information
                            </Text>
                                {(pendingEditRequest === false) &&
                                    <RouterLink className='misc-class-m-dropdown-link' to={`/profile/${profileId}/project`}>
                                        <Icon hover color={'contrast'} name={'edit'} width={1.5} height={1.5} />
                                    </RouterLink>
                                }
                            </Flex>
                            <ShadowBox p={3} key={profileJson.id} style={{ position: 'relative' }}>
                                <ProjectCard title={profileJson.name} textBody={profileJson.description} ministry={profileJson.ministryName} />
                            </ShadowBox>
                        </Box>
                        <Box>
                            <Flex p={3} mt={4} bg={theme.colors.bcblue} style={{ position: 'relative' }}>
                                <Text as="h3" color={theme.colors.contrast} mx={2} >
                                    Contact Information
                            </Text>
                                {(pendingEditRequest === false) &&
                                    <RouterLink className='misc-class-m-dropdown-link' to={`/profile/${profileId}/contact`}>
                                        <Icon hover color={'contrast'} name={'edit'} width={1.5} height={1.5} />
                                    </RouterLink>
                                }
                            </Flex>
                            <ShadowBox p={3} key={contactJson.id} style={{ position: 'relative' }}>
                                <ContactCard POName={contactJson.POName} POEmail={contactJson.POEmail} POGithubId={contactJson.POGithubId} TCName={contactJson.TCName} TCEmail={contactJson.TCEmail} TCGithubId={contactJson.TCGithubId} />
                            </ShadowBox>
                        </Box>
                        <Box>
                            <Flex p={3} mt={4} bg={theme.colors.bcblue} style={{ position: 'relative' }}>
                                <Text as="h3" color={theme.colors.contrast} mx={2} >
                                    Quota Information
                                </Text>
                                {(pendingEditRequest === false) &&
                                    <RouterLink className='misc-class-m-dropdown-link' to={`/profile/${profileId}/quota`}>
                                        <Icon hover color={'contrast'} name={'edit'} width={1.5} height={1.5} />
                                    </RouterLink>
                                }
                            </Flex>
                            <ShadowBox p={3} key={profileJson.id} style={{ position: 'relative' }}>
                                <QuotaCard licensePlate={licensePlate} quotaSize={quotaSize} />
                            </ShadowBox>
                        </Box>
                    </ShadowBox>
                </Box>
            </>
        );
    } else {
        return (
            <>
                <Flex p={3} mt={4} bg={theme.colors.bcblue}>
                    <RouterLink className='misc-class-m-dropdown-link' to={`/profile/${profileId}/overview`}>
                        <Icon hover color={'contrast'} name={'goBack'} width={1} height={1} />
                    </RouterLink>
                    <Text as="h3" color={theme.colors.contrast} mx={2} >
                        {(viewName)}
                    </Text>
                </Flex>
                <ShadowBox p={3}>
                    <Flex flexWrap='wrap' m={3}>
                        <ShadowBox p="24px" mt="0px" px={["24px", "24px", "70px"]} >
                            {(viewName === PROFILE_EDIT_VIEW_NAMES.PROJECT) &&
                                <ProjectCardEdit
                                    profileDetails={profileJson}
                                    ministry={ministry}
                                    openBackdropCB={openBackdropCB}
                                    closeBackdropCB={closeBackdropCB}
                                />
                            }
                            {(viewName === PROFILE_EDIT_VIEW_NAMES.CONTACT) &&
                                <ContactCardEdit
                                    profileId={profileId}
                                    contactDetails={contactJson}
                                    pendingEditRequest={pendingEditRequest}
                                    isProvisioned={provisionedStatus}
                                    setPendingEditRequest={setPendingEditRequest}
                                    openBackdropCB={openBackdropCB}
                                    closeBackdropCB={closeBackdropCB}
                                />
                            }
                            {(viewName === PROFILE_EDIT_VIEW_NAMES.QUOTA) &&
                                <QuotaCardEdit
                                    licensePlate={licensePlate}
                                    quotaSize={quotaSize}
                                    profileId={profileId}
                                    quotaOptions={quotaOptions}
                                    cnQuotaOptionsJson={cnQuotaOptionsJson}
                                    openBackdropCB={openBackdropCB}
                                    closeBackdropCB={closeBackdropCB}
                                    handleQuotaSubmitRefresh={handleQuotaSubmitRefresh}
                                    isProvisioned={isProvisioned}
                                />
                            }
                        </ShadowBox>
                        {(viewName === PROFILE_EDIT_VIEW_NAMES.QUOTA) && (
                            <Box p={"30px"} width={[1, 1, 1 / 3]}>
                                <Text>{txtForQuotaEdit}</Text>
                            </Box>
                        )}
                    </Flex>
                </ShadowBox>
            </>
        );
    }
};

export default ProfileEdit;
