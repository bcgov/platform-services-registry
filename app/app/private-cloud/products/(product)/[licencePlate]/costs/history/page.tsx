'use client';

import { Button } from '@mantine/core';
import { useQuery } from '@tanstack/react-query';
import { useEffect, useState } from 'react';
import { useSnapshot } from 'valtio';
import { z } from 'zod';
import LoadingBox from '@/components/generic/LoadingBox';
import FormYearPicker from '@/components/generic/select/FormYearPicker';
import YearlyCostBarChart from '@/components/private-cloud/charts/YearlyCostBarChart';
import { GlobalRole } from '@/constants';
import createClientPage from '@/core/client-page';
import { getTransformedCostData } from '@/helpers/product';
import {
  downloadPrivateCloudYearlyCosts,
  getPrivateCloudProductYearlyCostHistory,
} from '@/services/backend/private-cloud/products';
import { pageState } from './state';
import TableBody from './TableBody';

const pathParamSchema = z.object({
  licencePlate: z.string(),
});

const privateCloudProductCostHistory = createClientPage({
  roles: [GlobalRole.User],
  validations: { pathParams: pathParamSchema },
});

export default privateCloudProductCostHistory(({ getPathParams, session }) => {
  const [pathParams, setPathParams] = useState<z.infer<typeof pathParamSchema>>();
  const { year } = useSnapshot(pageState);

  const [downloading, setDownloading] = useState(false);

  useEffect(() => {
    getPathParams().then((v) => setPathParams(v));
  }, []);

  const { licencePlate = '' } = pathParams ?? {};

  const { data: data, isLoading: isLoading } = useQuery({
    queryKey: ['getPrivateCloudProductYearlyCostHistory', licencePlate, year],
    queryFn: () => getPrivateCloudProductYearlyCostHistory(licencePlate, year),
    enabled: !!licencePlate && !!year,
  });

  const setSelectedYear = (year: Date | null) => {
    if (year) {
      pageState.year = year.getFullYear().toString();
    } else {
      pageState.year = new Date().getFullYear().toString();
    }
  };
  const yearlyCostData = data?.items;
  const transformedData = getTransformedCostData(yearlyCostData || []);

  return (
    <>
      <FormYearPicker
        label="Select Year"
        value={new Date(parseInt(year), 0, 1)}
        onChange={(year) => setSelectedYear(year)}
        placeholder="Choose a year"
        defaultCurrentYear={true}
      />

      {yearlyCostData && (
        <LoadingBox isLoading={isLoading}>
          <>
            {yearlyCostData.length > 0 && (
              <div className="flex justify-end mb-4">
                <Button
                  color="success"
                  loading={downloading}
                  onClick={async () => {
                    setDownloading(true);
                    await downloadPrivateCloudYearlyCosts(licencePlate, year);
                    setDownloading(false);
                  }}
                >
                  Download PDF
                </Button>
              </div>
            )}

            <YearlyCostBarChart
              isLoading={isLoading}
              index="month"
              chartData={transformedData}
              title={`Cost History for ${year}`}
              categories={['CPU Cost', 'Storage Cost']}
            />
            <TableBody data={transformedData} currentYear={year} />
          </>
        </LoadingBox>
      )}
    </>
  );
});
