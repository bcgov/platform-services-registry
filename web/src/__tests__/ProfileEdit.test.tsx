//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import { render, waitFor } from '@testing-library/react';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Route, Router } from 'react-router-dom';
import ProfileEdit from '../views/ProfileEdit';
import mockContacts from './fixtures/profile-contacts.json';
import mockMinistry from './fixtures/profile-ministry.json';
import mockProfile from './fixtures/profiles.json';

const browserHistory = createBrowserHistory();

jest.mock(
  '../hooks/useRegistryApi',
  () =>
    function useRegistryApi() {
      const getProfileByProfileId = jest.fn().mockResolvedValue({
        data: [mockProfile],
      });

      const getMinistry = jest.fn().mockResolvedValue({
        data: [mockMinistry],
      });

      const getContactsByProfileId = jest.fn().mockResolvedValue({
        data: mockContacts,
      });

      return { getProfileByProfileId, getMinistry, getContactsByProfileId };
    },
);

jest.mock(
  '../utils/transformDataHelper',
  () =>
    function getProfileMinistry() {
      const mockGetProfileMinistry = jest.fn().mockResolvedValue({
        data: [
          {
            name: "Citizen's Services",
          },
        ],
      });
      return { mockGetProfileMinistry };
    },
);

jest.mock(
  '../utils/transformDataHelper',
  () =>
    function getProfileContacts() {
      const mockGetProfileContacts = jest.fn().mockResolvedValue({
        data: mockContacts,
      });
      return { mockGetProfileContacts };
    },
);

function renderProfileEdit() {
  const utils = render(
    <Router history={browserHistory}>
      <Route path="/profile/1/overview">
        <ProfileEdit />
      </Route>
    </Router>,
  );

  return { ...utils };
}

test('matches the snapshot', async () => {
  const { container } = renderProfileEdit();

  await waitFor(() => expect(container).toMatchSnapshot());
});
