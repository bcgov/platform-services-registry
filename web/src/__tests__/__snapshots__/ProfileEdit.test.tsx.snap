// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuotaCardEdit / > Card view should render 1`] = `<div />`;

exports[`<QuotaCardEdit / > Card view should render 2`] = `
<div>
  <h1
    class="css-zi8sy1"
  >
    License plates for the openshift namespaces
  </h1>
  <label
    class="css-1h8msqk"
    for="project-quotaCpuSize"
  >
    473f50
  </label>
  <br />
  <p
    class="css-1guafq4"
  >
    MEDIUM
     size quota for CPU:
    <br />
    8
     cores as request,
    16
     cores as limit
  </p>
  <br />
  <p
    class="css-1guafq4"
  >
    MEDIUM
     size quota for RAM:
    <br />
    32GiB
     as request, 
    64GiB
     as limit
  </p>
  <br />
  <p
    class="css-1guafq4"
  >
    MEDIUM
     size quota for storage: 20 PVC count,
    <br />
    100GiB
     overall storage with 
    50GiB
     for backup storage
  </p>
  <br />
  <select>
    <option>
      Select...
    </option>
    <option
      value="small"
    >
      small
    </option>
    <option
      value="large"
    >
      large
    </option>
  </select>
  <button
    class="css-s0jrgc"
    disabled=""
    style="display: block;"
  >
    Request Quota
  </button>
</div>
`;

exports[`matches the snapshot 1`] = `<div />`;
